Makefile

```diff
@@ -1,10 +1,65 @@
-test:
+# COLORS
+RED=\033[0;31m
+GREEN=\033[0;32m
+ORANGE=\033[0;33m
+BLUE=\033[0;34m
+PURPLE=\033[0;35m
+CYAN=\033[0;36m
+LIGHTGRAY=\033[0;37m
+DARKGRAY=\033[1;30m
+YELLOW=\033[0;33m
+NC=\033[0m # No Color
+
+chip?=F4
+BSPConfig?=off
+
+# Used to select the MCU (default is STM32F413)
+ifeq ($(chip), F4)
+    MCU_TARGET = STM32F413
+else ifeq ($(chip), L4)
+    MCU_TARGET = STM32L431
+else
+    $(error ${RED}Invalid MCU specified ${NC}(L4 and F4 are the only options))
+endif
+
+# to do: this is hella jank and only here since BSP.C is built for the F4 MCU. 
+# Keeping it optional to use BSP.C is nice, but there are cleaner ways we can do it.
+
+# If BSPConfig = on, we compile BSP.C
+ifeq ($(BSPConfig), on)
+    BSPConfig = on
+endif
+
+bsp_test:
 ifdef test
-	$(MAKE) -C BSP -C STM32F413 -j TARGET=bsp PROJECT_DIR=../.. BUILD_DIR=../../Objects TEST=../Tests/$(test).c
+	$(MAKE) -C BSP -C $(MCU_TARGET) -j TARGET=bsp PROJECT_DIR=../.. BUILD_DIR=../../Objects TEST=../Tests/$(test).c
 else
-	$(error test is not set (e.g. make test test=HelloWorld))
+	@echo -e "${RED}error${NC}test is not set (e.g. make bsp_test test=HelloWorld)"
 endif
 
+hw_test:
+ifdef test
+	$(MAKE) -C BSP -C $(MCU_TARGET) -j TARGET=bsp PROJECT_DIR=../.. BUILD_DIR=../../Objects TEST=../../HwTests/$(test).c
+else
+	@echo -e "${RED}error${NC}test is not set (e.g. make hw_test test=HelloWorld)"
+endif
+
```

Comment:
HwTests rlly only exists because originally the make test test=[name] target was set to only run tests in BSP/Tests, so I added a new thing to not interfere with that. I figured I should change that in a diff PR, but ig I can change it now.

Reply:
Time to brick every pr :D